### DISCUSSION TITLE
商品画像の取得方法について知りたい

### DISCUSSION THREAD

**Discussion by kenjiyoshid-a** [AlesInfiny/maia#1296] [05: Q&A] [answered]
# 質問
「api/catalog-items/」で新しいカタログアイテムをPOSTした際に商品画像（AssetCodes）はどのように処理されていますでしょうか。

# 質問のモチベーション
「api/catalog-items/{id}」でidによるカタログアイテムを取得する際に、AssetCodesも一緒に取得すると思うのですが、そもそもPOSTする際に画像データを登録していないためAssetCodesが取得できません。
どこかで何かしらデフォルトの画像をgetするような処理を実装されているのでしょうか。

**Comment by KentaHizume** [comment, marked as answer]
クライアント側で、画像がない場合（AssetCodesが空の配列だった場合）はNow Printingの画像を取得するように実装しています。

```env
# 商品画像が存在しない場合の代替画像の URL です。
VITE_NO_ASSET_URL=/api/assets/e622b0098808492cb883831c05486b58
```

```ts
export function assetHelper() {
  const getAssetUrl = (assetCode: string): string => {
    if (assetCode === '') {
      return `${import.meta.env.VITE_NO_ASSET_URL}`;
    }
```
